src/Http/Cookie/CookieCryptTrait.php

```diff
@@ -0,0 +1,194 @@
+    {
+        if (empty($this->encryptionKey)) {
+            return Security::salt();
+        }
+
+        return $this->encryptionKey;
+    }
+
+    /**
+     * Encrypts $value using public $type method in Security class
+     *
+     * @param string $value Value to encrypt
+     * @param string|bool $encrypt Encryption mode to use. False
+     *   disabled encryption.
+     * @param string|null $key Used as the security salt if specified.
+     * @return string Encoded values
+     */
+    protected function _encrypt($value, $encrypt, $key = null)
+    {
+        if (is_array($value)) {
+            $value = $this->_flatten($value);
+        }
+
+        $this->checkCipher($encrypt);
+        $prefix = 'Q2FrZQ==.';
+        $cipher = null;
+
+        if ($key === null) {
+            $key = $this->getCryptoKey();
+        }
+        if ($encrypt === 'rijndael') {
+            $cipher = Security::rijndael($value, $key, 'encrypt');
+        }
+        if ($encrypt === 'aes') {
+            $cipher = Security::encrypt($value, $key);
+        }
+
+        return $prefix . base64_encode($cipher);
+    }
+
+    /**
+     * Helper method for validating encryption cipher names.
+     *
+     * @param string $encrypt The cipher name.
+     * @return void
+     * @throws \RuntimeException When an invalid cipher is provided.
+     */
+    protected function checkCipher($encrypt)
+    {
+        if (!in_array($encrypt, $this->_validCiphers)) {
+            $msg = sprintf(
+                'Invalid encryption cipher. Must be one of %s.',
+                implode(', ', $this->_validCiphers)
+            );
+            throw new RuntimeException($msg);
+        }
+    }
+
+    /**
+     * Decrypts $value using public $type method in Security class
+     *
+     * @param array $values Values to decrypt
+     * @param string|bool $mode Encryption mode
+     * @param string|null $key Used as the security salt if specified.
+     * @return string|array Decrypted values
+     */
+    protected function _decrypt($values, $mode, $key = null)
+    {
+        if (is_string($values)) {
+            return $this->_decode($values, $mode, $key);
+        }
+
+        $decrypted = [];
+        foreach ($values as $name => $value) {
+            $decrypted[$name] = $this->_decode($value, $mode, $key);
+        }
+
+        return $decrypted;
+    }
+
+    /**
+     * Decodes and decrypts a single value.
+     *
+     * @param string $value The value to decode & decrypt.
+     * @param string|false $encrypt The encryption cipher to use.
+     * @param string|null $key Used as the security salt if specified.
+     * @return string|array Decoded values.
+     */
+    protected function _decode($value, $encrypt, $key)
+    {
+        if (!$encrypt) {
+            return $this->_expand($value);
+        }
+
+        $this->checkCipher($encrypt);
+        $prefix = 'Q2FrZQ==.';
+        $value = base64_decode(substr($value, strlen($prefix)));
```

Comment:
@burzum I might be wrong but I'm pretty sure that this function needs to be symetric with the _encode the this line should be moved after the if L191